File: MASTER DSA/SQUARES OF A SORTED ARRAY.java
/**
Squares of a Sorted Array |  Easy | Adobe |

Given an integer array nums sorted in non-decreasing order, return an array of the squares of each number sorted 
in non-decreasing order.

 

Example 1:

Input: nums = [-4,-1,0,3,10]
Output: [0,1,9,16,100]
Explanation: After squaring, the array becomes [16,1,0,9,100].
After sorting, it becomes [0,1,9,16,100].
 */

 
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] res = new int[nums.length];

        int i = nums.length - 1;
        int low = 0;
        int high = nums.length - 1;

        while (i >= 0) {
            if (Math.abs(nums[low]) > nums[high]) {
                res[i] = nums[low] * nums[low];
                low++;
            } else {
                res[i] = nums[high] * nums[high];
                high--;
            }
            i--;
        }

        return res;
    }
}

File: MASTER_DSA/1019-squares-of-a-sorted-array/squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        res = [0]*len(nums)


        left = 0
        right = len(nums)-1
        index = len(nums)-1

        while left<=right:
            # print(left,right)
            if abs(nums[left])>abs(nums[right]):
                res[index] = nums[left]*nums[left]
                left+=1
            else:
                res[index] = nums[right]*nums[right]
                right-=1

            index-=1

        return res
